src/components/calcite-color-hex-input/calcite-color-hex-input.tsx

```diff
@@ -113,45 +147,60 @@ export class CalciteColorHexInput {
 
   private onCalciteInputBlur = (event: Event): void => {
     const node = event.currentTarget as HTMLCalciteInputElement;
-    const hex = `#${node.value}`;
+    const inputValue = node.value;
+    const hex = `#${inputValue}`;
+    const willClearValue = this.allowEmpty && !inputValue;
 
-    if (isValidHex(hex) && isLonghandHex(hex)) {
+    if (willClearValue || (isValidHex(hex) && isLonghandHex(hex))) {
       return;
     }
 
+    // TODO: need to handle case where !valid hex and internal color is previously null - add test for this too!
```

Comment:
Is there an issue for this part or happening in this PR?

Reply:
Whoops! 😅 Good catch. It'll happen in this PR.

Reply:
💥